rules/Privatization/Rector/ClassMethod/PrivatizeFinalClassMethodRector.php

```diff
@@ -149,4 +169,38 @@ function (Node $node): bool {
 
         return $hasParentCall;
     }
+
+    private function shouldSkipClassMethodLaravel(ClassReflection $classReflection, ClassMethod $classMethod): bool
+    {
+        if (! $classReflection->is('Illuminate\Database\Eloquent\Model')) {
+            return false;
+        }
+
+        $name = (string) $this->getName($classMethod->name);
+        $returnType = $classMethod->returnType;
+
+        // Model attributes should be protected
+        if (
+            StringUtils::isMatch($name, self::LARAVEL_MODEL_ATTRIBUTE_REGEX)
+            || ($returnType instanceof Node && $this->isObjectType(
+                $returnType,
+                new ObjectType('Illuminate\Database\Eloquent\Casts\Attribute')
+            ))
+        ) {
+            return true;
+        }
+
+        // Model scopes should be protected
+        if (
+            StringUtils::isMatch($name, self::LARAVEL_MODEL_SCOPE_REGEX)
+            || $this->phpAttributeAnalyzer->hasPhpAttribute(
+                $classMethod,
+                'Illuminate\Database\Eloquent\Attributes\Scope'
+            )
+        ) {
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
Could you extract this to own service? So we have the fw and rule decoupled

Reply:
Somewhere around `rules/Privatization/NodeAnalyzer` e.g.

Reply:
Done